hooks/pre/31-upgrade-puppet.rb

```diff
@@ -74,5 +82,6 @@ def fail_and_exit(message)
   upgrade_step :remove_puppet_port_httpd
   upgrade_step :start_httpd
 
+  File.open('/etc/foreman-installer/.puppet_4_upgrade', 'w') { |file| file.write("completed on #{Time.now}") }
```

Comment:
Based on @ekohl's other comment I think we can change this (and the text below) to just say puppet upgrade to future-proof it so users can go directly from Puppet 3 to 5.